libvast/fbs/partition_synopsis.fbs

```diff
@@ -15,8 +16,36 @@ table LegacyPartitionSynopsis {
   import_time_range: vast.fbs.interval;
 }
 
+table FieldSketch {
+  field: type.detail.RecordField (required);
+  sketch: [ubyte] (required, nested_flatbuffer: "vast.fbs.Sketch");
+}
+
+table TypeSketch {
+  type: [ubyte] (required, nested_flatbuffer: "vast.fbs.Type");
+  sketch: [ubyte] (required, nested_flatbuffer: "vast.fbs.Sketch");
+}
+
+table PartitionSynopsisV1 {
+  layout_name: string;  
```

Comment:
Is there a specific reason why we're just storing a name rather than the full type of the partition here? Having the full type would make it possible to resolve extractors on it based on the synopsis alone, which makes many things easier in the catalog I think.